HerPublicWebsite.BusinessLogic/Services/RegularJobs/RegularJobsService.cs

```diff
@@ -45,4 +56,48 @@ public async Task RunNightlyTasksAsync()
             await dataProvider.PersistAllChangesAsync();
         }
     }
+
+    public async Task<IList<ReferralRequest>> GetReferralsPassedTenWorkingDayThresholdWithNoFollowUp()
+    {
+        var startDate = await AddWorkingDaysToDateTime(DateTime.Now, -10);
+        var newReferrals = await dataProvider.GetReferralRequestsWithNoFollowUpAfterDate(startDate);
+        return newReferrals;
+    }
+
+    public async Task<DateTime> AddWorkingDaysToDateTime(DateTime initialDateTime, int workingDaysToAdd)
+    {
+        int direction = workingDaysToAdd < 0 ? -1 : 1;
+        List<DateTime> holidays = await getHolidays();
+        DateTime newDateTime = initialDateTime;
+        while (workingDaysToAdd != 0)
+            {
+            newDateTime = newDateTime.AddDays(direction);
+            if (newDateTime.DayOfWeek != DayOfWeek.Saturday && 
+                newDateTime.DayOfWeek != DayOfWeek.Sunday && 
+                !holidays.Contains(newDateTime))
+            {
+                workingDaysToAdd -= direction;
+            }
+        }
+        return newDateTime;
+    }
+
+    private async Task<List<DateTime>> getHolidays() {
+            var parameters = new RequestParameters
+            {
+                BaseAddress = "https://www.gov.uk/bank-holidays.json",
+            };
+            var holidayDataJson = await HttpRequestHelper.SendGetRequestAsync<Dictionary<string, Holidays>>(parameters);
+            var englandAndWalesHolidays = holidayDataJson["england-and-wales"];
+            return englandAndWalesHolidays.events.Select(x => x.date).ToList();
+        }
+
+    private class Holidays {
+            public List<Event> events { get; set;}
```

Comment:
It should be upper case for the property names, so `Events` (and `Date` below).

This will still match up with the JSON API, but I think let's add a `JsonProperty` tag anyway. There are two reasons for that:

* It keeps the mapping explicit and deliberate.

* The static analysis in Rider is flagging that `set` as unused, which causes it to erroneously remove it when the class is cleaned (because the class is private and it can see nothing is setting it). Adding a `JsonProperty` attribute informs the static analysis that the property is used to decrypt JSON.

```
private class Holidays
{
    [JsonProperty("events")]
    public List<Event> Events { get; set; }
}
```

Same below for the `Event` class.